src/utils/flatten.js

```diff
@@ -55,12 +57,23 @@ const flatten = (
     /* Either execute or defer the function */
     if (typeof chunk === 'function') {
       if (executionContext) {
-        ruleSet.push(...flatten([chunk(executionContext)], executionContext))
+        ruleSet.push(
+          ...flatten([chunk(executionContext)], executionContext, styleSheet)
+        )
       } else ruleSet.push(chunk)
 
       return ruleSet
     }
 
+    if (chunk instanceof Keyframes) {
+      if (styleSheet) {
+        chunk.inject(styleSheet)
+        return ruleSet.concat(chunk.getName())
+      }
+
+      return ruleSet.concat(chunk)
```

Comment:
Looking at the function interpolation above, it seems that calling push and returning ruleSet is equally save and it has probably been changed by @probablyup as a micro optimisation, so let’s do the same here :)